File: Go/Add-Digits.go
package main

func addDigits(num int) int {
	if num < 10 {
		return num
	}

	num = sumOfDigits(num)

	return addDigits(num)

}

func sumOfDigits(num int) int {
	sum := 0

	for num > 0 {
		sum += num % 10
		num = num / 10
	}

	return sum
}

File: Rust/Add-Digits.rs
impl Solution {
    pub fn add_digits(num: i32) -> i32 {
        if num < 10 {
            return num;
        }

        Solution::add_digits(Solution::sum_of_digits(num))
    }

    pub fn sum_of_digits(mut num: i32) -> i32 {
        let mut sum = 0;

        while num > 0 {
            sum += num % 10;
            num /= 10;
        }

        sum
    }
}
